syncer/data_fetch.go

```diff
@@ -51,23 +52,25 @@ type (
 
 // DataFetch contains the logic of fetching mesh data.
 type DataFetch struct {
-	logger  log.Log
-	msh     meshProvider
-	fetcher fetcher
+	fetcher
+
+	logger log.Log
+	msh    meshProvider
 }
 
 // NewDataFetch creates a new DataFetch instance.
 func NewDataFetch(msh meshProvider, fetch fetcher, lg log.Log) *DataFetch {
 	return &DataFetch{
+		fetcher: fetch,
 		logger:  lg,
 		msh:     msh,
-		fetcher: fetch,
 	}
 }
 
 // PollLayerData polls all peers for data in the specified layer.
-func (d *DataFetch) PollLayerData(ctx context.Context, lid types.LayerID) error {
-	peers := d.fetcher.GetPeers()
+func (d *DataFetch) PollLayerData(ctx context.Context, lid types.LayerID, peers ...p2p.Peer) error {
+	peers = append(peers, d.fetcher.GetPeers()...)
```

Comment:
where fetcher gets peers? it is from currently open connections? where do you get peers that are sent to this method?

Reply:
they are also from fetcher.. this doesn't makes sense then